org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseProjectImporter.java

```diff
@@ -43,6 +48,21 @@ public boolean applies(IProgressMonitor monitor) throws CoreException {
 		if (directories == null) {
 			BasicFileDetector eclipseDetector = new BasicFileDetector(rootFolder.toPath(), DESCRIPTION_FILE_NAME)
 					.addExclusions("**/bin");//default Eclipse build dir
+			List<IProjectImporter> importers = new ArrayList<>(ProjectsManager.importers());
+			for (IProject project : ProjectUtils.getAllProjects(false)) {
+				File projectFile = project.getLocation().toFile();
+				Iterator<IProjectImporter> iter = importers.iterator();
+				while (iter.hasNext()) {
+					IProjectImporter importer = iter.next();
+					if (!(importer instanceof EclipseProjectImporter)) {
+						importer.initialize(projectFile);
+						if (importer.applies(monitor)) {
```

Comment:
I'm cautious about using importer.applies here. The applies is a heavy operation, and usually it will scan the root folder recursively. When iterating the importers list, it means each importer will scan the root again.

Reply:
I have updated the PR. Please review.